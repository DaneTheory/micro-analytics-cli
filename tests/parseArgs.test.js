jest.mock('pkginfo', () => () => ({version: '1.0.0'}))
const parseArgs = require('../src/parseArgs')

describe('parseArgs', () => {
  it("should have correct defaults", () => {
    expect(parseArgs(['node', 'micro-analytics'])).toMatchSnapshot();
  })

  it("should use DB_ADAPTER environment variable as default if set", () => {
    process.env.DB_ADAPTER = 'redis'
    expect(parseArgs(['node', 'micro-analytics']).adapter).toEqual('redis');
    delete process.env.DB_ADAPTER
  })

  it("should use PORT environment variable as default if set", () => {
    process.env.PORT = '3000'
    expect(parseArgs(['node', 'micro-analytics']).port).toEqual(3000);
    delete process.env.PORT
  })

  it("should use HOST environment variable as default if set", () => {
    process.env.HOST = 'localhost'
    expect(parseArgs(['node', 'micro-analytics']).host).toEqual('localhost');
    delete process.env.HOST
  })

  it('should use get adapter option when using -a', () => {
    process.env.DB_ADAPTER = 'redis'
    const args = ['node', 'micro-analytics', '-a', 'flat-file-db']
    expect(Object.keys(parseArgs(args))).toContain('dbName');
    delete process.env.DB_ADAPTER
  });

  it('should use get adapter option when using --adapter', () => {
    process.env.DB_ADAPTER = 'redis'
    const args = ['node', 'micro-analytics', '--adapter', 'flat-file-db']
    expect(Object.keys(parseArgs(args))).toContain('dbName');
    delete process.env.DB_ADAPTER
  });

  it('should use get adapter option when using --adapter=', () => {
    process.env.DB_ADAPTER = 'redis'
    const args = ['node', 'micro-analytics', '--adapter=flat-file-db']
    expect(Object.keys(parseArgs(args))).toContain('dbName');
    delete process.env.DB_ADAPTER
  });
})
