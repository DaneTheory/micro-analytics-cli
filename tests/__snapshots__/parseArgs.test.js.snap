// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parseArgs should have correct defaults 1`] = `
Object {
  "H": "0.0.0.0",
  "a": "flat-file-db",
  "adapter": "flat-file-db",
  "d": "views.db",
  "dbName": "views.db",
  "host": "0.0.0.0",
  "p": 3000,
  "port": 3000,
}
`;
